aptos-move/aptos-vm/src/move_vm_ext/write_op_converter.rs

```diff
@@ -170,16 +169,22 @@ impl<'r> WriteOpConverter<'r> {
         move_storage_op: MoveStorageOp<BytesWithResourceLayout>,
         legacy_creation_as_modification: bool,
     ) -> PartialVMResult<(WriteOp, Option<Arc<MoveTypeLayout>>)> {
-        let result = self.convert(
-            self.remote.get_resource_state_value_metadata(state_key),
-            move_storage_op.clone(),
+        let state_value_metadata = self
```

Comment:
why did we change the order here, is that intentional? does this change behavior?

Reply:
it doesn't change the behaviour, but avoids cloning the write op, which also makes `self.convert` cleaner